source/tomopy/prep/stripe.py

```diff
@@ -939,3 +941,151 @@ def _remove_stripe_based_interpolation(tomo, snr, size, drop_ratio, norm):
+    radius : int, optional
+        The pixel size of the stencil to calculate the mean ratio between vertical and horizontal orientations. The larger values will enlarge the mask width.
+    ncore : int, optional
+        Number of cores that will be assigned to jobs. All cores will be used
+        if unspecified.
+
+    Returns
+    -------
+    ndarray
+        Weights for stripe's edges as a 3D array of float32 data type. 
+        The weights can be thresholded or passed to stripes_mask3d function to obtain a binary mask.
+
+    Raises
+    ------
+    ValueError
+        If the input array is not three dimensional.        
+    """ 
+    if ncore is None:
+        ncore = mproc.mp.cpu_count()
+
+    input_type = tomo.dtype
+    if (input_type != 'float32'):
+        tomo = dtype.as_float32(tomo)  # silent convertion to float32 data type
+    out = np.empty_like(tomo, order='C')
+    
+    if tomo.ndim == 3:
+        dz, dy, dx = tomo.shape
+        if (dz == 0) or (dy == 0) or (dx == 0):
+            raise ValueError("The length of one of dimensions is equal to zero")
+    else:
+        raise ValueError("The input array must be a 3D array")
+
+    if size <= 0 or size > dz //  2:
+        raise ValueError("The size of the filter should be larger than zero and smaller than the half of the vertical dimension")
+
+    # perform stripes detection
+    extern.c_stripes_detect3d(np.ascontiguousarray(tomo), 
+                              out, 
+                              size,
+                              radius,
+                              ncore,
+                              dx, dy, dz)
+    return out
+
+
+def stripes_mask3d(weights, 
+                  threshold = 0.6,
+                  min_stripe_length = 20,
+                  min_stripe_depth  = 10,
+                  min_stripe_width = 5,
+                  sensitivity_perc = 85.0,
+                  ncore=None):
+    """
+    Takes the result of the stripes_detect3d module as an input and generates a 
+    binary 3D mask with ones where stripes present. The method tries to eliminate
+    non-stripe features in data by checking the weight consistency in three directions.
+
+    .. versionadded:: 1.14
+
+    Parameters
+    ----------
+    weights : ndarray
+        3D weights array, a result of stripes_detect3d module given in
+        [angles, detY(depth), detX] axis orientation.
+    threshold : float, optional
+        Threshold for the given weights, the smaller values correspond to the stripes
+    min_stripe_length : int, optional
+        Minimum accepted length of a stripe in pixels. Can be large if there are full stripes in the data.
+    min_stripe_depth : int, optional
+        Minimum accepted depth of a stripe in pixels. The stripes do not extend very deep, with this parameter more non-stripe features can be removed. 
+    min_stripe_width : int, optional
```

Comment:
Although I understand the logic of why they should have the axis title in the name it can be that `length`, `depth` and `width` are more the stripe-related descriptors rather than the axis related? I've put the axis names in the description, let me know if it works for you?

Reply:
This was helpful, but there is still double naming of axes which are confusing.

For example, "The pixel size of the vertical (angle dimension) " I think this statement means to imply that the angle dimension of the `tomo` array is also the vertical dimension, but ND-arrays don't have a vertical dimension, they have a 0th, 1st, 2nd, 3rd, ... dimensions.

Instead of "With this orientation, the stripes are the vertical features.", we should say "With this orientation, the stripes are features in axis 0" or "With this orientation, the stripes are features along the angle axis."

Reply:
OK, I think I understand your point here. I guess we're talking about two things - the algorithmic part and the data I/O part. I'm happy to go along with more abstract naming for dimensions and, as you suggest, refer to the data dimensions angle, detY, detX and not their spatial orientations with respect to data. Would you mind changing this or I can do it? thanks

Reply:
also corrected.